src/components/app/details/cdDetails/CDDetails.tsx

```diff
@@ -143,18 +144,25 @@ export default function CDDetails(){
         }
     },[deploymentHistoryResult, loadingDeploymentHistory, deploymentHistoryError])
 
-      
     useEffect(() => {
+        setLoader(true);
         if(pipelineId){
         try {
-                getDeploymentTemplateDiff(appId, pipelineId).then((response) => {
-                    setDeploymentTemplatesConfiguration(response.result.sort((a, b) => sortCallback('id', b, a)));
-                });
+            getDeploymentTemplateDiff(appId, pipelineId).then((response) => {
+                setDeploymentTemplatesConfiguration(response.result.sort((a, b) => sortCallback('id', b, a)));
+                setLoader(false);
+            });
+
         } catch (err) {
             showError(err);
+            setLoader(false);
         }
-    }
-    }, [pipelineId]);
+
+        return (): void => {
+            if (showTemplate) {
+                setShowTemplate(false);
```

Comment:
Remove this as we don't need this in CDDetails.

Reply:
done